alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -242,121 +205,131 @@ class DocumentLevelMonitorRunner : MonitorRunner() {
-            if (monitor.triggers.isEmpty()) {
-                if (dryrun == false && monitor.id != Monitor.NO_ID) {
-                    createFindings(monitor, monitorCtx, docsToQueries, idQueryMap, true)
-                }
-            } else {
-                monitor.triggers.forEach {
-                    triggerResults[it.id] = runForEachDocTrigger(
+                    val nodeMap = getNodes(monitorCtx)
+                    val nodeShardAssignments = distributeShards(
                         monitorCtx,
-                        monitorResult,
-                        it as DocumentLevelTrigger,
-                        monitor,
-                        idQueryMap,
-                        docsToQueries,
-                        queryToDocIds,
-                        dryrun,
-                        executionId = executionId,
-                        workflowRunContext = workflowRunContext
+                        nodeMap.keys.toList(),
+                        shards.toList(),
+                        concreteIndexName
                     )
+
+                    val responses: Collection<DocLevelMonitorFanOutResponse> = suspendCoroutine { cont ->
+                        val listener = GroupedActionListener(
+                            object : ActionListener<Collection<DocLevelMonitorFanOutResponse>> {
+                                override fun onResponse(response: Collection<DocLevelMonitorFanOutResponse>) {
+                                    cont.resume(response)
+                                }
+
+                                override fun onFailure(e: Exception) {
+                                    if (e.cause is Exception)
+                                        cont.resumeWithException(e.cause as Exception)
+                                    else
+                                        cont.resumeWithException(e)
+                                }
+                            },
+                            nodeShardAssignments.size
+                        )
+                        val responseReader = Writeable.Reader {
+                            DocLevelMonitorFanOutResponse(it)
+                        }
+                        for (node in nodeMap) {
+                            if (nodeShardAssignments.containsKey(node.key)) {
+                                val docLevelMonitorFanOutRequest = DocLevelMonitorFanOutRequest(
+                                    monitor,
+                                    dryrun,
+                                    monitorMetadata,
+                                    executionId,
+                                    indexExecutionContext,
+                                    nodeShardAssignments[node.key]!!.toList(),
+                                    concreteIndicesSeenSoFar,
+                                    workflowRunContext
+                                )
+
+                                transportService.sendRequest(
```

Comment:
Do we need a rate limiting mechanism to avoid overwhelming the cluster? Theoretically this could submit hundreds of requests in parallel which could consume all of the cluster's resources until execution completes. This would interrupt other processes like indexing or user search traffic